test/Grpc.NetCore.HttpClient.Tests/AsyncDuplexStreamingCallTests.cs

```diff
@@ -0,0 +1,166 @@
+            {
+                content = (PushStreamContent)request.Content;
+                await content.PushComplete.DefaultTimeout();
+
+                return ResponseUtils.CreateResponse(HttpStatusCode.OK, new StreamContent(streamContent));
+            });
+            var invoker = new HttpClientCallInvoker(httpClient);
+
+            // Act
+            var call = invoker.AsyncDuplexStreamingCall<HelloRequest, HelloReply>(TestHelpers.ServiceMethod, null, new CallOptions());
+
+            var requestStream = call.RequestStream;
+            var responseStream = call.ResponseStream;
+
+            // Assert
+            await call.RequestStream.WriteAsync(new HelloRequest { Name = "1" }).DefaultTimeout();
+            await call.RequestStream.WriteAsync(new HelloRequest { Name = "2" }).DefaultTimeout();
+
+            await call.RequestStream.CompleteAsync().DefaultTimeout();
+
+            var requestContent = await content.ReadAsStreamAsync().DefaultTimeout();
+            var requestMessage = await requestContent.ReadStreamedMessageAsync(TestHelpers.ServiceMethod.RequestMarshaller.Deserializer, CancellationToken.None).DefaultTimeout();
+            Assert.AreEqual("1", requestMessage.Name);
+            requestMessage = await requestContent.ReadStreamedMessageAsync(TestHelpers.ServiceMethod.RequestMarshaller.Deserializer, CancellationToken.None).DefaultTimeout();
+            Assert.AreEqual("2", requestMessage.Name);
+
+            Assert.IsNull(responseStream.Current);
+
+            var moveNextTask1 = responseStream.MoveNext(CancellationToken.None);
+            Assert.IsFalse(moveNextTask1.IsCompleted);
+
+            await streamContent.AddDataAndWait(await TestHelpers.GetResponseDataAsync(new HelloReply
+            {
+                Message = "Hello world 1"
+            }).DefaultTimeout()).DefaultTimeout();
+
+            Assert.IsTrue(await moveNextTask1.DefaultTimeout());
+            Assert.IsNotNull(responseStream.Current);
+            Assert.AreEqual("Hello world 1", responseStream.Current.Message);
+
+            var moveNextTask2 = responseStream.MoveNext(CancellationToken.None);
+            Assert.IsFalse(moveNextTask2.IsCompleted);
+
+            await streamContent.AddDataAndWait(await TestHelpers.GetResponseDataAsync(new HelloReply
+            {
+                Message = "Hello world 2"
+            }).DefaultTimeout()).DefaultTimeout();
+
+            Assert.IsTrue(await moveNextTask2.DefaultTimeout());
+            Assert.IsNotNull(responseStream.Current);
+            Assert.AreEqual("Hello world 2", responseStream.Current.Message);
+
+            var moveNextTask3 = responseStream.MoveNext(CancellationToken.None);
+            Assert.IsFalse(moveNextTask3.IsCompleted);
+
+            await streamContent.AddDataAndWait(Array.Empty<byte>()).DefaultTimeout();
+
+            Assert.IsFalse(await moveNextTask3.DefaultTimeout());
+            Assert.IsFalse(await moveNextTask3.DefaultTimeout());
```

Comment:
Why the second assert

Reply:
The contract is that once MoveNext() returns false or throws, it will continue doing so forever. Add a comment?